### PR TITLE
Add .asin() method to Lua documentation

### LINKED ISSUE #3387: [Term Entry] Lua Math .asin()
### Reason/inspiration (optional)

We would like a new term entry on the `.asin()` method for Lua: [Math](https://www.codecademy.com/resources/docs/lua/mathematical-library). The entry should go in a new file under `docs/content/lua/concepts/mathematical-library/terms/asin/asin.md`.

The entry should describe what the method does. Be sure to include the following sections:

- A "Syntax" section with a ` ```pseudo` code snippet.
- An "Example" section with a ` ```lua `code snippet.

Please refer to the [entry templates](https://github.com/Codecademy/docs/blob/main/documentation/), [content standards](https://github.com/Codecademy/docs/blob/main/documentation/content-standards.md), and [markdown style guide](https://curriculum-documentation.codecademy.com/content-guidelines/markdown-style-guide/) when working on the PR for this issue.

### Entry Type (select all that apply)

New Term for Existing Concept Entry

### Code of Conduct

- [X] By submitting this issue, I agree to follow Codecademy Doc's [Code of Conduct](https://github.com/Codecademy/docs/blob/main/.github/CODE_OF_CONDUCT.md).

### For Maintainers

- [X] Labels added
- [ ] Issue is assigned

### PR DESCRIPTION
**Pull request by laisvigas**
### Description

Closes #3387

The document provides information about Lua's math.asin() function, explaining its usage and practical applications, and I have created a new entry for this topic in the documentation.

### Type of Change

- Adding a new entry
- Updating the documentation

### Checklist

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [x] Under "Development" on the right, I have linked any issues that are relevant to this PR (write "Closes #<issue number> in the "Description" above).

### CODE REVIEW

File: content/lua/concepts/mathematical-library/terms/asin/asin.md [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+  - 'Math'
+  - 'Numbers'
+CatalogContent:
+  - 'learn-lua'
+  - 'paths/computer-science'
+---
+
+Lua's `math.asin()` function calculates the inverse sine, also known as arcsine, of a given value. In other words, it finds the angle in radians whose sine equals the given value. The function takes a number between -1 and 1 as input and returns the corresponding angle in radians. This is particularly useful in various applications, such as physics simulations and game development, where understanding angles and trajectories is essential.
+
+> **Note**: Mastering the intricacies of `math.asin()` is crucial for anyone in the programming field. Although specific tasks might be accomplished through libraries and frameworks, a solid understanding of how the function works is fundamental. It not only aids in solving immediate problems but also empowers programmers to develop new applications and innovate in their domains.
```
Comment (dakshdeepHERE) [inline content/lua/concepts/mathematical-library/terms/asin/asin.md:19]:
```suggestion
> **Note**: Understanding the inner workings of `math.asin()` is essential for anyone in the programming field. While specific tasks can often be achieved using libraries and frameworks, a solid grasp of this function's functionality is fundamental. This knowledge not only helps in solving immediate problems but also empowers programmers to create new applications and innovate within their respective domains.
```

File: content/lua/concepts/mathematical-library/terms/asin/asin.md [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+```shell
+1.5707963267949
+```
+
+## Example 2
+
+The Lua script below provides a deeper example of `math.asin()`. It calculates and prints the launch angle in degrees required to hit a target 15 meters away using an arrow traveling at a speed of 10 meters per second, demonstrating the use of the `math.asin()` function in trigonometric calculations for projectile motion.
+
+```lua
+-- Fixed values
```
Comment (dakshdeepHERE) [inline content/lua/concepts/mathematical-library/terms/asin/asin.md:50]:
```suggestion
```

File: content/lua/concepts/mathematical-library/terms/asin/asin.md [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+local arrow_speed = 10 -- speed of the arrow in meters per second
+local target_distance = 15 -- distance to the target in meters
+
+-- Calculating launch angle using simplified formula
```
Comment (dakshdeepHERE) [inline content/lua/concepts/mathematical-library/terms/asin/asin.md:54]:
```suggestion
```

**Review by dakshdeepHERE — CHANGES_REQUESTED** [review]
Some changes from my side. Will look through it again once you commit these.

**Comment by laisvigas** [comment]
@dakshdeepHERE Done.

**Comment by dakshdeepHERE** [comment]
@laisvigas LGTM🚀

**Review by dakshdeepHERE — APPROVED** [review]

File: content/lua/concepts/mathematical-library/terms/asin/asin.md [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+  - 'Math'
+  - 'Numbers'
+CatalogContent:
+  - 'learn-lua'
+  - 'paths/computer-science'
+---
+
+Lua's **`math.asin()`** function calculates the inverse sine, also known as arcsine, of a given value. In other words, it finds the angle in radians whose sine equals the given value. The function takes a number within the range of `-1` to `1` as input, representing a sine value, and returns the corresponding angle in radians. This is particularly useful in various applications, such as physics simulations and game development, where understanding angles and trajectories is essential for tasks like character movement and object positioning.
+
+> **Note**: Understanding the inner workings of `math.asin()` is essential for anyone in the programming field. While specific tasks can often be achieved using libraries and frameworks, a solid grasp of this function's functionality is fundamental. This knowledge not only helps in solving immediate problems but also empowers programmers to create new applications and innovate within their respective domains.
```
Comment (caupolicandiaz) [inline content/lua/concepts/mathematical-library/terms/asin/asin.md:19]:
```suggestion
```

Comment (caupolicandiaz) [reply content/lua/concepts/mathematical-library/terms/asin/asin.md:19]: This note is not really bringing attention to a specific aspect of the functionality, or a caveat to be aware of. It's more of a general recommendation how understanding the implementation can have broader benefits in understanding other programming problems.

File: content/lua/concepts/mathematical-library/terms/asin/asin.md [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+
+## Syntax
+
+The `math.asin()` function must be called with an argument using the following syntax:
+
+```pseudo
+math.asin(x)
+```
+
+- Where `x` is a number between - 1 and 1, representing a sine value for finding the corresponding angle in radians. If no argument is passed, the program will raise an error.
```
Comment (caupolicandiaz) [inline content/lua/concepts/mathematical-library/terms/asin/asin.md:29]:
```suggestion
- Where `x` is a number between `-1` and `1`, representing a sine value for finding the corresponding angle in radians. If no argument is passed, the program will raise an error.
```

File: content/lua/concepts/mathematical-library/terms/asin/asin.md [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+print(math.asin(1))
+```
+
+Output:
+
+```shell
+1.5707963267949
+```
+
+## Example 2
```
Comment (caupolicandiaz) [inline content/lua/concepts/mathematical-library/terms/asin/asin.md:45]: This example doesn't seem quite right. Generally, projectile problems (whether time of flight, max height, etc) all include a gravitational component (g), which is not present here. I would suggest deleting the example for the sake of simplification

**Review by caupolicandiaz — CHANGES_REQUESTED** [review]
@laisvigas I've added my comments. Please update the entry, and tag me when you've finished making changes. Thanks

**Comment by laisvigas** [comment]
@caupolicandiaz Done.

**Review by caupolicandiaz — APPROVED** [review]
@laisvigas This looks good

**Pull request merged**